modules/artifact/src/main/java/org/mule/runtime/module/artifact/api/classloader/MuleArtifactClassLoader.java

```diff
@@ -50,9 +50,14 @@
   static final Pattern DOT_REPLACEMENT_PATTERN = Pattern.compile("\\.");
   static final String PATH_SEPARATOR = "/";
   static final String RESOURCE_PREFIX = "resource::";
-  static final String WILDCARD = "*";
-
-  private static final String NO_WILDCARD_NO_SPACES = "([^\\" + WILDCARD + "|\\s]+)";
+  static final String WILDCARD = "\\*";
+
+  private static final String SPACES = "\\s";
+  private static final String RESOURCE_CAPTURE_START = "([^";
+  private static final String PIPE = "|";
+  private static final String RESOURCE_CAPTURE_END = "]+)";
+  private static final String RESOURCE = RESOURCE_CAPTURE_START + WILDCARD + PIPE + RESOURCE_CAPTURE_END;
```

Comment:
Why is the pipe needed in this case? Sounds like you just need a NO_WILDCARDS regex.